File: Problem10/euler.py
def is_prime(n):
    if n == 2 or n == 3: return True
    if n < 2 or n%2 == 0: return False
    if n < 9: return True
    if n%3 == 0: return False
    r = int(n**0.5)
    f = 5
    while f <= r:
        if n%f == 0: return False
        if n%(f+2) == 0: return False
        f +=6
    return True  

list = []

list.append(2)
for k in range(3, 1000001, 2):
    if is_prime(k):
        list.append(k)

for i in range(int(raw_input())):
    n = int(raw_input())
    i, sum =0, 0
    while(True):
        if list[i] < n+1:
            sum = sum + list[i]
            i = i + 1
        else:
            break
    print sum

File: Problem10/euler.cpp
#include <cmath>
#include <cstring>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;
typedef unsigned long long int ll;
ll prime_sum[1000001];

bool isPrime(ll num) {
    if(num < 2) {
        return false;
    } else if(num < 4) {
        return true;
    } else if(num % 2 == 0 || num % 3 == 0) {
        return false;
    }
    for(int i = 3, max = sqrt(num); i <= max; i += 2) {
        if(num % i == 0) {
            return false;
        }
    }
    return true;
}

void find_all_sum()
{
    ll sum = 0;
    ll limit = 1000001;
    for(ll i=0; i < limit; i++)
    {
        if(isPrime(i))
        {
            sum += i;
        }
        prime_sum[i] = sum;
        //cout<<i<<" "<<prime_sum[i]<<endl;
    }
}

int main() {
    ll testcases, n;
    cin>>testcases;
    find_all_sum();
    while(testcases--)
    {
        cin>>n;
        cout<<prime_sum[n]<<endl;
    }
    return 0;
}